File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
   for(let i=0;i<= haystack.length - needle.length;i++){

    if(haystack.substring(i,i+needle.length) === needle){
        return i
    }
   }

 if(haystack === needle){
    return 0
 }

    return -1
};

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        haystack_length = len(haystack)
        needle_length = len(needle)
      
        for start_index in range(haystack_length - needle_length + 1):
            if haystack[start_index:start_index + needle_length] == needle:
                return start_index
      
        return -1